System/Random.hs

```diff
@@ -430,22 +433,55 @@ mkStdGen32 sMaybeNegative = StdGen (s1+1) (s2+1)
 createStdGen :: Integer -> StdGen
 createStdGen s = mkStdGen32 $ fromIntegral s
 
-{- |
-With a source of random number supply in hand, the 'Random' class allows the
-programmer to extract random values of a variety of types.
 
-Minimal complete definition: 'randomR' and 'random'.
-
--}
+-- $uniform
+--
+-- Type classes 'Uniform' and 'UniformRange' are unrelated type
+-- classes. This is deliberate decision because there're types which
+-- could have instance for one type class but not the other.
+--
+-- For example: 'Integer', 'Float', 'Double' have instance for
+-- @UniformRange@ but there's no way to define @Uniform@.
+--
+-- Conversely there're types where @Uniform@ instance is possible
+-- while @UniformRange@ is not. One example is:
+--
+-- > instance (Uniform a, Uniform b) => Uniform (a,b)
+--
+-- This instance is very straightforward. @UniformRange@ isn't very
```

Comment:
I don't understand this comment.

Reply:
I meant that even if we have `(UniformRange a, UniformRange b)` we can't write instance for `UniformRange (a,b)`. But looking back this part is very poorly written. I'll rewrite it